src/VecSim/spaces/IP_space.cpp

```diff
@@ -64,6 +64,15 @@ dist_func_t<double> IP_FP64_GetDistFunc(size_t dim) {
     case ARCH_OPT_NONE:
         break;
     case ARCH_OPT_AVX512:
+#ifdef __AVX512F__
+    {
+        static dist_func_t<double> dist_funcs[] = {
+            FP64_InnerProduct, FP64_InnerProductSIMD8Ext_AVX512, FP64_InnerProductSIMD2Ext_AVX512,
+            FP64_InnerProductSIMD8ExtResiduals_AVX512, FP64_InnerProductSIMD2ExtResiduals_AVX512};
+
+        ret_dist_func = dist_funcs[optimization_type];
```

Comment:
How come we don't get coverage here?

Reply:
Basically we didn't have coverage in tests_spaces for the `spaces::<space>_<data_type>_GetDistFunc` functions (`<space>` is L2 or IP, `<data_type>` is FP32 or FP64), so basically this should be covered only in tests that are creating an index with FP64 vectors (@meiravgri PR). But I think it is better to test it here also, so I added a check to verify that the chosen distance function is expected given the `dim` in every test.